src/io/metadata.cpp

```diff
@@ -253,11 +258,40 @@ void Metadata::LoadInitialScore() {
 
   Log::Info("Start loading initial scores");
   num_init_score_ = static_cast<data_size_t>(reader.Lines().size());
-  init_score_ = new float[num_init_score_];
+
+  if (num_init_score_ <= 0){
+      return ;
+  }
+  
+  std::vector<std::string> oneline_init_score = Common::Split(reader.Lines()[0].c_str(), '\t');
+  num_class_ = static_cast<int>(oneline_init_score.size());
+  
+  init_score_ = new score_t[num_init_score_ * num_class_];
   double tmp = 0.0f;
-  for (data_size_t i = 0; i < num_init_score_; ++i) {
-    Common::Atof(reader.Lines()[i].c_str(), &tmp);
-    init_score_[i] = static_cast<float>(tmp);
+  
+  if (num_class_ == 1){
+      for (data_size_t i = 0; i < num_init_score_; ++i) {
+        Common::Atof(reader.Lines()[i].c_str(), &tmp);
+        init_score_[i] = static_cast<score_t>(tmp);
```

Comment:
@chivee I think, class by class is preferred, because it reads from or writes to successive addresses. But here, the scores are parsed sample by sample.